src/components/NewCardForm.js

```diff
@@ -3,4 +3,63 @@ import PropTypes from 'prop-types';
 import emoji from 'emoji-dictionary';
 import './NewCardForm.css';
 
-const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]
+const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"];
+
+class NewCardForm extends Component {
+  constructor(props) {
+    super(props);
+
+    const resetState = {
+      text: '',
+      emoji: ''
+    };
+
+    this.state = {...resetState};
+    this.resetState = {...resetState}
+  }
+
+  onFieldChange = (event) => {
+    const { name, value } = event.target;
+    const updatedField = {};
+    updatedField[name] = value;
+    this.setState(updatedField);
+  };
+
+  submitForm = (event) => {
+    event.preventDefault();
+    const newCard = {...this.state};
+    this.props.addCardCallback(newCard);
+    this.setState(this.resetState);
+  };
+
+  getEmojiList = () => {
+  const options = EMOJI_LIST.map((text, index) => {
+    const emojiUnicode = emoji.getUnicode(text);
+    return <option key={ index } value={ text }>{ emojiUnicode }</option>
```

Comment:
Good work breaking this out as a helper function - it makes this code very clear.